### DISCUSSION TITLE
Language support in the WPF Editor

### DISCUSSION THREAD

**Discussion by RobertvanderHulst** [VsixCommunity/Community.VisualStudio.Toolkit#356] [General]
For our custom language (X#) we are trying to complete the language support in the WPF Editor.
We already have an EventBindingProvider, so when you double click a control in the designer window a method is generated or the existing method is opened.
However when you are in the source code subwindow of the designer I would like to implement Peek Definition and Goto Definition support.
The C# editor allows this and goto definition on attributes of controls even goes to the decompiled source, and goto definition on the name of an event goes to the C# code in the project.
I tried to add a ProvideLanguageExtension attribute for xaml files but when I do that then we are apparently expected to do all the handling on the window, including colorization and dropdown comboboxes.
I would like to only add support for goto definition and peek definition for methods in our project. 
Ideally all the other stuff, such as the colorization, dropdown comboboxes, goto definition for attributes of control definitions should work like in C# (it does not work now).
Does anybody know how to implement this?
Does anybody know if the C# implementation for this is open source?
I looked in the C# project system and in the Roslyn source but could not find this. Maybe I looked for the wrong words?

**Comment by mrlacey** [comment]
As I'm sure you're aware, the official WPF repo is at https://github.com/dotnet/wpf **but** like with most of the Windows-related open-source projects from Microsoft, all the integration with Visual Studio is not open-source :(

FYI. possibly also of interest: https://github.com/microsoft/xaml-designer-extensibility
